hack/kube-utils.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+# Copyright 2023 Red Hat, Inc. and/or its affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+getKubeSystemPodStatusConditions() { 
+  kubectl get pods $1 -n kube-system -o json | jq '.items[].status.conditions[]'
+}
+
+getKubeSystemPodReadyStatus() { 
+  echo $(kubectl get pods $1 -n kube-system -o json | jq -r '.items[].status.conditions[] | select(.type == "Ready") | .status') 
+}
+
+export -f getKubeSystemPodStatusConditions
+export -f getKubeSystemPodReadyStatus
```

Comment:
FYI export is not needed if you are sourcing. Export will just make it available in the child scripts, if there are any. I mean, it is not really necessary here. However, this way we could use them in some scripts called from the minikube.groovy script, for example.